src/applications/vaos/reducers/appointments.js

```diff
@@ -64,6 +66,27 @@ export default function appointmentsReducer(state = initialState, action) {
         futureStatus: FETCH_STATUS.failed,
         future: null,
       };
+    case FETCH_REQUEST_MESSAGES_SUCCEEDED: {
+      let future = state.future;
+      if (future.length > 0 && action.messages?.length > 0) {
+        const messages = action.messages.sort(sortMessages);
+        future = future.map(a => {
+          const current = a;
+          if (
+            current.appointmentRequestId &&
+            current.appointmentRequestId === action.requestId
+          ) {
+            current.messages = messages;
```

Comment:
I think you'd want to do `current = {...current, messages }`, so you don't modify the `a` object.

Reply:
Another thing to consider would be putting these in a separate object in the state, using the request ids as keys. Then you could pass the right list of messages as a prop in the components. I'm doing something like that for the facility information that we also fetch separately.